crates/codegen/parser/runtime/src/napi_interface/cst.rs

```diff
@@ -68,10 +68,45 @@ impl RuleNode {
             .into()
     }
 
+    #[napi(catch_unwind, js_name = "toJSON")]
+    /// Serialize the token node to JSON.
+    ///
+    /// This method is intended for debugging purposes and may not be stable.
+    pub fn to_json(&self) -> String {
+        serde_json::to_string(&self.0).unwrap()
+    }
+
     #[napi(catch_unwind)]
     pub fn unparse(&self) -> String {
         self.0.clone().unparse()
     }
+
+    // Expose the children as a hidden (non-enumerable, don't generate type definition)
+    // property that's eagerly evaluated (getter) in the debugger context.
+    #[napi(
+        enumerable = false,
```

Comment:
Getters are eagerly evaluated by vscode-js-debug and chrome devtools. The property is needed since otherwise we would not be able to present and allow for convenient tree navigation, as seen in the screenshots in the OP. The second post here explains that we do *not* use custom formatters with a thorough explanation why, so there's no other way to present this in the debugger rather than via a property.